### DISCUSSION TITLE
Lexicon version 1: How specifically are the `.env` files used?

### DISCUSSION THREAD

**Discussion by Ashistalin06** [lexiconhq/lexicon#31] [Lexicon Version 1] [answered]
### Describe the bug
I would like to know which files are to be changes in order to make the build working?

For instance, like we change the MOBILE_PROSE_HOST to where we have deployed the code in .env file from frontend folder. 
Similarly, do we need to edit .env from API folder as well, if yes what changes? 
And package.json etc, any changes required in them?

### *REQUIRED:* Include the contents of `api/.env` and `frontend/.env`
_If you are manually overriding any environment variables when running the package scripts, include those as well._

**api/.env**
```
PROSE_DISCOURSE_HOST=https://meta.discourse.org
PROSE_APP_HOSTNAME=0.0.0.0
PROSE_APP_PORT=8929
```

**frontend/.env**
```
MOBILE_PROSE_HOST=https://meta.discourse.org
```

### _REQUIRED:_ answer the following questions:
- If this is related to your mobile app:
  - Are you running it through Expo, or did you build it?
  
I was running through metro and I could see it was getting connected to the server and fetching all the channels, posts etc.

  - Is it running on your mobile device or on a simulator?

Mobile device.

### To Reproduce
1. Download and install all the dependencies of lexicon
2. Run the installed lexicon app.
3. In development machine it is working fine and displaying all the details of channels, posts users etc.
4. I used `eas build` command and created an apk. After installing it in my mobile, I am getting 'Connect to a network' issue. 
6. Expo has mentioned something about .env variables into their eas.json, I have added value 'https://meta.discourse.org' in extra of app.json. Still I am not able to get connected to the server.

### Expected behavior
I want the app to run like it is running in development machine. 

### Environments
Windows

#### Desktop (please complete the following information):
 - OS: Windows
 - Browser: Crome
 - Version: 106.0.5249.119

#### Smartphone (please complete the following information):
 - Device: Nothing phone 1, Realme x2 pro
 - OS: NothingOS, ColorOS
 - Browser : Crome
 - Version : 106.0.5249.126

#### GraphQL API (please complete the following information):
Please indicate where the Prose GraphQL API is running.
- 0.0.0.0:80

Are you only running it locally on your development machine?
- development machine 

Have you deployed it somewhere? If so, what domain or IP address did you deploy it at?
- I am using MOBILE_PROSE_HOST=https://meta.discourse.org

**Comment by Kinostrome** [comment]
Please do not skip filling out the issue template. We'll need to close this issue unless you update your post with the issue template fully filled out.

**Comment by Ashistalin06** [comment]
I have edited and added the issue template.

My query is, in FRONTED folder, in package.json file there is quickstart which is given value of "MOBILE_PROSE_HOST=http://127.0.0.1 MOBILE_PROSE_PORT=8929 expo r -c".

While creating a build do I need to changes these also?

**Comment by Kinostrome** [comment, marked as answer]
If you want to publish your app, you'll need to deploy the API somewhere publicly accessible.

This topic was also recently covered in this issue: https://github.com/lexiconhq/lexicon/issues/11

I would recommend reading the documentation links mentioned in that issue for next steps.

Once you've deployed it somewhere, you will need to update the `frontend/.env` file to point at the correct host and port.

The line you're referencing from `package.json` is a **quick start** command used in development only.

When building the app you should follow what is outlined in the documentation under the Publishing your App section.

**Comment by Ashistalin06** [comment]
I'm having trouble with the build. The API I'm using is working in development mode only but not after creating a build. The API endpoint is: https://www.websitetoolbox.com/tool/view/mb/forum_search?q=forums.thebrain.com

Since `expo build:android` is going to depreciate in January 2023, I'm using command `eas build`.

Also, The API link I mentioned above is returning a JSON response. 
If I point my `MOBILE_PROSE_HOST` to that link above, that's only going to give me a JSON response. How do I make it return graphQL playground? 

I went through the doc and it said I have to host my lexicon code and the dependencies plus the port to that link above. Is that what I to do to solve this thing up?

**Comment by Kinostrome** [comment]
> Since expo build:android is going to depreciate in January 2023, I'm using command eas build.

The docs do not indicate to use eas build, so there should be no expectation that things will work as expected if you're not following the instructions as they're described.

> How do I make it return graphQL playground?

That depends on how you hosted it. Lexicon doesn't return responses like that so you'll have to debug your individual setup to figure out what's going wrong. Also, the docs recommend getting a dedicated instance to host it on, which does not appear to be the case here, so your mileage may vary.

> I went through the doc and it said I have to host my lexicon code and the dependencies plus the port to that link above. Is that what I to do to solve this thing up?

Yes, follow the instructions in the docs to deploy the GraphQL API somewhere publicly accsessible.

**Comment by Ashistalin06** [comment]
>Yes, follow the instructions in the docs to deploy the GraphQL API somewhere publicly accsessible.

The document didn't mention about setting up frontend/.env on the server just like we do in development mode. 
I feel like I'm confused, in short I have to install everything including api/.env but not frontend/.env part on the server, right?

**Comment by Kinostrome** [comment]
> The document didn't mention about setting up frontend/.env on the server just like we do in development mode.

This seems to communicate some misunderstanding. The frontend has nothing to do with the server.

The frontend is used to build the app.
The API needs to be deployed on a server.

Please read the documentation closely in the linked sections above to understand how to deploy the GraphQL API.

If you're still confused after that, I would suggest finding someone who is more familiar with NodeJS, React Native, deploying applications, etc. who can guide you through this.
